### ISSUE TITLE
EDSC-1468: EDSC orders are stuck in the 'creating' state possibly due to clogged up order queue due to large numbers of LAADS orders

### ISSUE THREAD

**Issue by earthdata-tasktop** [nasa/edsc-search-test-config-3#395]
Sudha and I placed orders with EDSC today and both saw the above problem.  
Example:  
sourcetype=prod *ngap* app_search-ops 6364594906

Solution: provide a way to clear out queue by user and/or provider. It should not involve ssh'ing to the host.

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

This was resolved with the linked NGAP ticket.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

From [~tgoff]:

It is speculation, however, we know this:

1) LAADS AND USGS_EROS provider orders are being suspended due to the firwall issue.

2) Orders for other providers are getting through

3) Orders submitted via OPS EDSC do not make it to legacy services(in a reasonable time period), but orders from OPS reverb or UAT EDSC pointed at OPS do

4) Orders submitted via OPS EDSC do eventually make it to legacy services, after a number of hours  

5) A single user is submitting an unusual volume of orders for LAADS data

Based on this, I think the best explanation we have is that the queue is being exhausted by the LAADS orders from a single user, and the suspension of LAADS order is contributing to its slower workoff (though Im not sure how this would happen, unless EDSC is checking on the status of orders it submits and only allowing a certain number to be in progress at a time).

**Comment by earthdata-tasktop** [comment]
(Comment from Sudha Murthy):

saw the issue in all three daacs - NSIDC/LPDAAC/LARC where the order was stuck in 'Creating'

**Comment by earthdata-tasktop** [comment]
(Comment from Sudha Murthy):

saw the issue in all three daacs - NSIDC/LPDAAC/LARC where the order was stuck in 'Creating'

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

I was able to reproduce the issue using the same production collection (MODIS/Terra Calibrated Radiances 5-Min L1B Swath 250m V006) in UAT. The retrieval id is [https://search.uat.earthdata.nasa.gov/data/retrieve/6895050843](https://search.uat.earthdata.nasa.gov/data/retrieve/6895050843)

The order stuck in "Creating" state because the order in CMR is in "SUBMITTING" state:

    curl -X GET \
      'https://cmr.earthdata.nasa.gov/legacy-services/rest/orders.json?id%5B%5D=9D37C739-ADC2-E124-4A55-417685C1A1AF%1B' \
      -H 'cache-control: no-cache' \
      -H 'echo-token: ???'

    [
      {
        "order": {
          "client_identity": "HoOcKmts_X6VtUtDYxNmbQ",
          "created_at": "2017-05-09T14:37:29Z",
          "id": "9D37C739-ADC2-E124-4A55-417685C1A1AF",
          "notification_level": "VERBOSE",
          "order_price": 0,
          "owner_id": "C3421B30-444F-23B4-03C6-5B756C4B4461",
          "provider_orders": [
            {
              "reference": {
                "id": "LAADS",
                "location": "https://cmr.earthdata.nasa.gov:/legacy-services/rest/orders/9D37C739-ADC2-E124-4A55-417685C1A1AF/provider_orders/LAADS",
                "name": "LAADS"
              }
            }
          ],
          "state": "SUBMITTING",
          "submitted_at": "2017-05-09T14:37:40Z",
          "updated_at": "2017-05-09T14:37:40Z",
          "user_domain": "GOVERNMENT",
          "user_region": "INTERNATIONAL"
        }
      }
    ]

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

From [~tgoff]:

    web.3ab39f9  | Failed to route provider order [E86B9865-C5D4-814C-CBF9-F4A4200E7B43] in order [01B2388D-DA4F-1995-15C9-C088A8BED1B6] due to an order transmission communication failure.Provider Id:[LAADS], Provider Guid:[56860]
    web.3ab39f9  | [mErrorCode:[OrderTransmissionCommFailure], mErrorInstanceId:[0138B76F-9D3C-3D82-BFE2-D40BB3A40A1E], mTimeStamp:[Tue May 09 14:42:32 UTC 2017], systemMessage:[Could not connect to the provider's order fulfillment service.], cause:[org.springframework.ws.client.WebServiceIOException: I/O error: Connection refused; nested exception is java.net.ConnectException: Connection refused]]
    web.3ab39f9  |     at gov.nasa.echo.kernel.service.order.routing.orderfulfillment.v9.OrderFulfillmentClientImpl.handleError(OrderFulfillmentClientImpl.java:235)
    web.3ab39f9  |     at gov.nasa.echo.kernel.service.order.routing.orderfulfillment.v9.OrderFulfillmentClientImpl.submit(OrderFulfillmentClientImpl.java:142)
    web.3ab39f9  |     at gov.nasa.echo.kernel.service.order.routing.orderfulfillment.v9.OrderFulfillmentStrategy.routeSubmit(OrderFulfillmentStrategy.java:188)
    web.3ab39f9  |     at gov.nasa.echo.kernel.service.order.routing.ProviderOrderRouterImpl.routeOrder(ProviderOrderRouterImpl.java:117)

Seems like LAADS is not accepting orders

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

All my orders submitted from OPS EDSC were not reaching CMR because a user has been submitting orders every 2 minutes day and night since this Monday. I can, however, do it from UAT EDSC pointing to OPS CMR, which makes me think that maybe all my orders were enqueued in OPS. So I would like to check the delayed jobs queue in ops from rails console.

irb(main):001:0> Delayed::Job.all.size  

=> 635  

irb(main):002:0> Delayed::Job.where(attempts: 0).size  

=> 365

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

Could be either an NGAP issue or CMR issue.

**Comment by earthdata-tasktop** [comment]
(Comment from Frank Schaffer):

This issue is affecting both NSIDC orders and EGI requests in Prod.

**Comment by earthdata-tasktop** [comment]
(Comment from Sudha Murthy):

saw the issue in all three daacs - NSIDC/LPDAAC/LARC where the order was stuck in 'Creating'

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

What caused the delayed queue backup?

* We don’t know what caused the backup. We suspect that it is one user who kept submitting orders around the clock since this Monday using collection C203234448-LAADS. Starting from 5/8 4:12pm to 7pm, the user submitted one order approximately every 5 minutes. From 7pm, the user brought up another instance to submit one order every 5 minutes and have been doing that since then. 
* Doug saw the issue this morning and Tim and I started investigating. Here are what we found:  
* LAADS has a firewall issue and all their orders are blocked. From the EDSC’s point of view, the order stuck in “Creating” state. (The one that the user used is a LAADS collection)  
* Using an NSIDC collection and submitting a new order in EDSC OPS, the order was not created in CMR db.  
* Using an NSIDC collection and submitting a new order in EDSC UAT pointing to CMR OPS (cmr_env=prod), the order reached CMR and was processed correctly.  
* Using an NSIDC collection and submitting a new order in EDSC dev from my laptop, the order was created in CMR db. (although it was not processed due to firewall issues for obvious reasons)  
* Delayed_job workers are busy in OPS  
* Delayed_job “queue” has a length of 365 (Delayed::Job.where(attempts: 0).size)  
* Every time the user submits an order, the order GUID is different, meaning that the queue of the delayed_job is moving.  
* When the queue is empty, the worker will log something like: 1 jobs processed at 8.5161 j/s, 0 failed. Splunk search sourcetype=prod*ngap*app_search-ops "jobs processed at” shows that the queue has never been empty starting from Monday. Also note that the processing time is 8.5s per job (I admit there are faster ones but the time varies greatly from provider to provider). If that is the case for LAADS or if the processing time is larger than the interval that the user submits, the queue will get longer and longer. 365 is the number from today around 3pm.
* All the findings are pointing to the delayed job queue being to long. However, I am not 100% sure that cleaning up the queue will resolve this issue.

I suggest to recycle both ops nodes first and take another look at the delayed_jobs status:  

A) Find all rows in edsc db: Delayed::Job.all.size  

B) Find queued rows (i.e. the size of the queue): Delayed::Job.where(attempts: 0).size  

C) Find failed jobs (which are permanently kept in the db; successful ones are immediately removed from the db upon completion): Delayed::Job.where.not(failed_at: nil).size  

A) = B) + C)

Only clean up the queue if the size gets larger over time.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

The short term solution is 

* purge the db
* notify the user to stop submitting repeated orders to EDSC
* add status banner to indicate that LAADS ordering is down

Long term: we could increase the number of workers to work off the queue faster. (in file: earthdata-search/Procfile, add -n=2 before run in the second line)

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

From [~tgoff]:

It is speculation, however, we know this:

1) LAADS AND USGS_EROS provider orders are being suspended due to the firwall issue.

2) Orders for other providers are getting through

3) Orders submitted via OPS EDSC do not make it to legacy services(in a reasonable time period), but orders from OPS reverb or UAT EDSC pointed at OPS do

4) Orders submitted via OPS EDSC do eventually make it to legacy services, after a number of hours  

5) A single user is submitting an unusual volume of orders for LAADS data

Based on this, I think the best explanation we have is that the queue is being exhausted by the LAADS orders from a single user, and the suspension of LAADS order is contributing to its slower workoff (though Im not sure how this would happen, unless EDSC is checking on the status of orders it submits and only allowing a certain number to be in progress at a time).

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[~dnewman], do you concur that this ticket is no longer needed?

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

From [~tgoff]:

    web.3ab39f9  | Failed to route provider order [E86B9865-C5D4-814C-CBF9-F4A4200E7B43] in order [01B2388D-DA4F-1995-15C9-C088A8BED1B6] due to an order transmission communication failure.Provider Id:[LAADS], Provider Guid:[56860]
    web.3ab39f9  | [mErrorCode:[OrderTransmissionCommFailure], mErrorInstanceId:[0138B76F-9D3C-3D82-BFE2-D40BB3A40A1E], mTimeStamp:[Tue May 09 14:42:32 UTC 2017], systemMessage:[Could not connect to the provider's order fulfillment service.], cause:[org.springframework.ws.client.WebServiceIOException: I/O error: Connection refused; nested exception is java.net.ConnectException: Connection refused]]
    web.3ab39f9  |     at gov.nasa.echo.kernel.service.order.routing.orderfulfillment.v9.OrderFulfillmentClientImpl.handleError(OrderFulfillmentClientImpl.java:235)
    web.3ab39f9  |     at gov.nasa.echo.kernel.service.order.routing.orderfulfillment.v9.OrderFulfillmentClientImpl.submit(OrderFulfillmentClientImpl.java:142)
    web.3ab39f9  |     at gov.nasa.echo.kernel.service.order.routing.orderfulfillment.v9.OrderFulfillmentStrategy.routeSubmit(OrderFulfillmentStrategy.java:188)
    web.3ab39f9  |     at gov.nasa.echo.kernel.service.order.routing.ProviderOrderRouterImpl.routeOrder(ProviderOrderRouterImpl.java:117)

Seems like LAADS is not accepting orders

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

All my orders submitted from OPS EDSC were not reaching CMR because a user has been submitting orders every 2 minutes day and night since this Monday. I can, however, do it from UAT EDSC pointing to OPS CMR, which makes me think that maybe all my orders were enqueued in OPS. So I would like to check the delayed jobs queue in ops from rails console.

irb(main):001:0> Delayed::Job.all.size  

=> 635  

irb(main):002:0> Delayed::Job.where(attempts: 0).size  

=> 365

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

Could be either an NGAP issue or CMR issue.

**Comment by earthdata-tasktop** [comment]
(Comment from Frank Schaffer):

This issue is affecting both NSIDC orders and EGI requests in Prod.

**Comment by earthdata-tasktop** [comment]
(Comment from Sudha Murthy):

saw the issue in all three daacs - NSIDC/LPDAAC/LARC where the order was stuck in 'Creating'

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

What caused the delayed queue backup?

* We don’t know what caused the backup. We suspect that it is one user who kept submitting orders around the clock since this Monday using collection C203234448-LAADS. Starting from 5/8 4:12pm to 7pm, the user submitted one order approximately every 5 minutes. From 7pm, the user brought up another instance to submit one order every 5 minutes and have been doing that since then. 
* Doug saw the issue this morning and Tim and I started investigating. Here are what we found:  
* LAADS has a firewall issue and all their orders are blocked. From the EDSC’s point of view, the order stuck in “Creating” state. (The one that the user used is a LAADS collection)  
* Using an NSIDC collection and submitting a new order in EDSC OPS, the order was not created in CMR db.  
* Using an NSIDC collection and submitting a new order in EDSC UAT pointing to CMR OPS (cmr_env=prod), the order reached CMR and was processed correctly.  
* Using an NSIDC collection and submitting a new order in EDSC dev from my laptop, the order was created in CMR db. (although it was not processed due to firewall issues for obvious reasons)  
* Delayed_job workers are busy in OPS  
* Delayed_job “queue” has a length of 365 (Delayed::Job.where(attempts: 0).size)  
* Every time the user submits an order, the order GUID is different, meaning that the queue of the delayed_job is moving.  
* When the queue is empty, the worker will log something like: 1 jobs processed at 8.5161 j/s, 0 failed. Splunk search sourcetype=prod*ngap*app_search-ops "jobs processed at” shows that the queue has never been empty starting from Monday. Also note that the processing time is 8.5s per job (I admit there are faster ones but the time varies greatly from provider to provider). If that is the case for LAADS or if the processing time is larger than the interval that the user submits, the queue will get longer and longer. 365 is the number from today around 3pm.
* All the findings are pointing to the delayed job queue being to long. However, I am not 100% sure that cleaning up the queue will resolve this issue.

I suggest to recycle both ops nodes first and take another look at the delayed_jobs status:  

A) Find all rows in edsc db: Delayed::Job.all.size  

B) Find queued rows (i.e. the size of the queue): Delayed::Job.where(attempts: 0).size  

C) Find failed jobs (which are permanently kept in the db; successful ones are immediately removed from the db upon completion): Delayed::Job.where.not(failed_at: nil).size  

A) = B) + C)

Only clean up the queue if the size gets larger over time.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

The short term solution is 

* purge the db
* notify the user to stop submitting repeated orders to EDSC
* add status banner to indicate that LAADS ordering is down

Long term: we could increase the number of workers to work off the queue faster. (in file: earthdata-search/Procfile, add -n=2 before run in the second line)

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

From [~tgoff]:

It is speculation, however, we know this:

1) LAADS AND USGS_EROS provider orders are being suspended due to the firwall issue.

2) Orders for other providers are getting through

3) Orders submitted via OPS EDSC do not make it to legacy services(in a reasonable time period), but orders from OPS reverb or UAT EDSC pointed at OPS do

4) Orders submitted via OPS EDSC do eventually make it to legacy services, after a number of hours  

5) A single user is submitting an unusual volume of orders for LAADS data

Based on this, I think the best explanation we have is that the queue is being exhausted by the LAADS orders from a single user, and the suspension of LAADS order is contributing to its slower workoff (though Im not sure how this would happen, unless EDSC is checking on the status of orders it submits and only allowing a certain number to be in progress at a time).

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[~dnewman], do you concur that this ticket is no longer needed?

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

All my orders submitted from OPS EDSC were not reaching CMR because a user has been submitting orders every 2 minutes day and night since this Monday. I can, however, do it from UAT EDSC pointing to OPS CMR, which makes me think that maybe all my orders were enqueued in OPS. So I would like to check the delayed jobs queue in ops from rails console.

irb(main):001:0> Delayed::Job.all.size  

=> 635  

irb(main):002:0> Delayed::Job.where(attempts: 0).size  

=> 365

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

Could be either an NGAP issue or CMR issue.

**Comment by earthdata-tasktop** [comment]
(Comment from Frank Schaffer):

This issue is affecting both NSIDC orders and EGI requests in Prod.

**Comment by earthdata-tasktop** [comment]
(Comment from Sudha Murthy):

saw the issue in all three daacs - NSIDC/LPDAAC/LARC where the order was stuck in 'Creating'

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

What caused the delayed queue backup?

* We don’t know what caused the backup. We suspect that it is one user who kept submitting orders around the clock since this Monday using collection C203234448-LAADS. Starting from 5/8 4:12pm to 7pm, the user submitted one order approximately every 5 minutes. From 7pm, the user brought up another instance to submit one order every 5 minutes and have been doing that since then. 
* Doug saw the issue this morning and Tim and I started investigating. Here are what we found:  
* LAADS has a firewall issue and all their orders are blocked. From the EDSC’s point of view, the order stuck in “Creating” state. (The one that the user used is a LAADS collection)  
* Using an NSIDC collection and submitting a new order in EDSC OPS, the order was not created in CMR db.  
* Using an NSIDC collection and submitting a new order in EDSC UAT pointing to CMR OPS (cmr_env=prod), the order reached CMR and was processed correctly.  
* Using an NSIDC collection and submitting a new order in EDSC dev from my laptop, the order was created in CMR db. (although it was not processed due to firewall issues for obvious reasons)  
* Delayed_job workers are busy in OPS  
* Delayed_job “queue” has a length of 365 (Delayed::Job.where(attempts: 0).size)  
* Every time the user submits an order, the order GUID is different, meaning that the queue of the delayed_job is moving.  
* When the queue is empty, the worker will log something like: 1 jobs processed at 8.5161 j/s, 0 failed. Splunk search sourcetype=prod*ngap*app_search-ops "jobs processed at” shows that the queue has never been empty starting from Monday. Also note that the processing time is 8.5s per job (I admit there are faster ones but the time varies greatly from provider to provider). If that is the case for LAADS or if the processing time is larger than the interval that the user submits, the queue will get longer and longer. 365 is the number from today around 3pm.
* All the findings are pointing to the delayed job queue being to long. However, I am not 100% sure that cleaning up the queue will resolve this issue.

I suggest to recycle both ops nodes first and take another look at the delayed_jobs status:  

A) Find all rows in edsc db: Delayed::Job.all.size  

B) Find queued rows (i.e. the size of the queue): Delayed::Job.where(attempts: 0).size  

C) Find failed jobs (which are permanently kept in the db; successful ones are immediately removed from the db upon completion): Delayed::Job.where.not(failed_at: nil).size  

A) = B) + C)

Only clean up the queue if the size gets larger over time.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

The short term solution is 

* purge the db
* notify the user to stop submitting repeated orders to EDSC
* add status banner to indicate that LAADS ordering is down

Long term: we could increase the number of workers to work off the queue faster. (in file: earthdata-search/Procfile, add -n=2 before run in the second line)

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

From [~tgoff]:

It is speculation, however, we know this:

1) LAADS AND USGS_EROS provider orders are being suspended due to the firwall issue.

2) Orders for other providers are getting through

3) Orders submitted via OPS EDSC do not make it to legacy services(in a reasonable time period), but orders from OPS reverb or UAT EDSC pointed at OPS do

4) Orders submitted via OPS EDSC do eventually make it to legacy services, after a number of hours  

5) A single user is submitting an unusual volume of orders for LAADS data

Based on this, I think the best explanation we have is that the queue is being exhausted by the LAADS orders from a single user, and the suspension of LAADS order is contributing to its slower workoff (though Im not sure how this would happen, unless EDSC is checking on the status of orders it submits and only allowing a certain number to be in progress at a time).

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[~dnewman], do you concur that this ticket is no longer needed?

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673

**Comment by earthdata-tasktop** [comment]
(Comment from Frank Schaffer):

This issue is affecting both NSIDC orders and EGI requests in Prod.

**Comment by earthdata-tasktop** [comment]
(Comment from Sudha Murthy):

saw the issue in all three daacs - NSIDC/LPDAAC/LARC where the order was stuck in 'Creating'

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

What caused the delayed queue backup?

* We don’t know what caused the backup. We suspect that it is one user who kept submitting orders around the clock since this Monday using collection C203234448-LAADS. Starting from 5/8 4:12pm to 7pm, the user submitted one order approximately every 5 minutes. From 7pm, the user brought up another instance to submit one order every 5 minutes and have been doing that since then. 
* Doug saw the issue this morning and Tim and I started investigating. Here are what we found:  
* LAADS has a firewall issue and all their orders are blocked. From the EDSC’s point of view, the order stuck in “Creating” state. (The one that the user used is a LAADS collection)  
* Using an NSIDC collection and submitting a new order in EDSC OPS, the order was not created in CMR db.  
* Using an NSIDC collection and submitting a new order in EDSC UAT pointing to CMR OPS (cmr_env=prod), the order reached CMR and was processed correctly.  
* Using an NSIDC collection and submitting a new order in EDSC dev from my laptop, the order was created in CMR db. (although it was not processed due to firewall issues for obvious reasons)  
* Delayed_job workers are busy in OPS  
* Delayed_job “queue” has a length of 365 (Delayed::Job.where(attempts: 0).size)  
* Every time the user submits an order, the order GUID is different, meaning that the queue of the delayed_job is moving.  
* When the queue is empty, the worker will log something like: 1 jobs processed at 8.5161 j/s, 0 failed. Splunk search sourcetype=prod*ngap*app_search-ops "jobs processed at” shows that the queue has never been empty starting from Monday. Also note that the processing time is 8.5s per job (I admit there are faster ones but the time varies greatly from provider to provider). If that is the case for LAADS or if the processing time is larger than the interval that the user submits, the queue will get longer and longer. 365 is the number from today around 3pm.
* All the findings are pointing to the delayed job queue being to long. However, I am not 100% sure that cleaning up the queue will resolve this issue.

I suggest to recycle both ops nodes first and take another look at the delayed_jobs status:  

A) Find all rows in edsc db: Delayed::Job.all.size  

B) Find queued rows (i.e. the size of the queue): Delayed::Job.where(attempts: 0).size  

C) Find failed jobs (which are permanently kept in the db; successful ones are immediately removed from the db upon completion): Delayed::Job.where.not(failed_at: nil).size  

A) = B) + C)

Only clean up the queue if the size gets larger over time.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

The short term solution is 

* purge the db
* notify the user to stop submitting repeated orders to EDSC
* add status banner to indicate that LAADS ordering is down

Long term: we could increase the number of workers to work off the queue faster. (in file: earthdata-search/Procfile, add -n=2 before run in the second line)

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

From [~tgoff]:

It is speculation, however, we know this:

1) LAADS AND USGS_EROS provider orders are being suspended due to the firwall issue.

2) Orders for other providers are getting through

3) Orders submitted via OPS EDSC do not make it to legacy services(in a reasonable time period), but orders from OPS reverb or UAT EDSC pointed at OPS do

4) Orders submitted via OPS EDSC do eventually make it to legacy services, after a number of hours  

5) A single user is submitting an unusual volume of orders for LAADS data

Based on this, I think the best explanation we have is that the queue is being exhausted by the LAADS orders from a single user, and the suspension of LAADS order is contributing to its slower workoff (though Im not sure how this would happen, unless EDSC is checking on the status of orders it submits and only allowing a certain number to be in progress at a time).

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[~dnewman], do you concur that this ticket is no longer needed?

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

The short term solution is 

* purge the db
* notify the user to stop submitting repeated orders to EDSC
* add status banner to indicate that LAADS ordering is down

Long term: we could increase the number of workers to work off the queue faster. (in file: earthdata-search/Procfile, add -n=2 before run in the second line)

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

From [~tgoff]:

It is speculation, however, we know this:

1) LAADS AND USGS_EROS provider orders are being suspended due to the firwall issue.

2) Orders for other providers are getting through

3) Orders submitted via OPS EDSC do not make it to legacy services(in a reasonable time period), but orders from OPS reverb or UAT EDSC pointed at OPS do

4) Orders submitted via OPS EDSC do eventually make it to legacy services, after a number of hours  

5) A single user is submitting an unusual volume of orders for LAADS data

Based on this, I think the best explanation we have is that the queue is being exhausted by the LAADS orders from a single user, and the suspension of LAADS order is contributing to its slower workoff (though Im not sure how this would happen, unless EDSC is checking on the status of orders it submits and only allowing a certain number to be in progress at a time).

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[~dnewman], do you concur that this ticket is no longer needed?

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

From [~tgoff]:

It is speculation, however, we know this:

1) LAADS AND USGS_EROS provider orders are being suspended due to the firwall issue.

2) Orders for other providers are getting through

3) Orders submitted via OPS EDSC do not make it to legacy services(in a reasonable time period), but orders from OPS reverb or UAT EDSC pointed at OPS do

4) Orders submitted via OPS EDSC do eventually make it to legacy services, after a number of hours  

5) A single user is submitting an unusual volume of orders for LAADS data

Based on this, I think the best explanation we have is that the queue is being exhausted by the LAADS orders from a single user, and the suspension of LAADS order is contributing to its slower workoff (though Im not sure how this would happen, unless EDSC is checking on the status of orders it submits and only allowing a certain number to be in progress at a time).

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[~dnewman], do you concur that this ticket is no longer needed?

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[~dnewman], do you concur that this ticket is no longer needed?

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

Could be either an NGAP issue or CMR issue.

**Comment by earthdata-tasktop** [comment]
(Comment from Frank Schaffer):

This issue is affecting both NSIDC orders and EGI requests in Prod.

**Comment by earthdata-tasktop** [comment]
(Comment from Sudha Murthy):

saw the issue in all three daacs - NSIDC/LPDAAC/LARC where the order was stuck in 'Creating'

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

What caused the delayed queue backup?

* We don’t know what caused the backup. We suspect that it is one user who kept submitting orders around the clock since this Monday using collection C203234448-LAADS. Starting from 5/8 4:12pm to 7pm, the user submitted one order approximately every 5 minutes. From 7pm, the user brought up another instance to submit one order every 5 minutes and have been doing that since then. 
* Doug saw the issue this morning and Tim and I started investigating. Here are what we found:  
* LAADS has a firewall issue and all their orders are blocked. From the EDSC’s point of view, the order stuck in “Creating” state. (The one that the user used is a LAADS collection)  
* Using an NSIDC collection and submitting a new order in EDSC OPS, the order was not created in CMR db.  
* Using an NSIDC collection and submitting a new order in EDSC UAT pointing to CMR OPS (cmr_env=prod), the order reached CMR and was processed correctly.  
* Using an NSIDC collection and submitting a new order in EDSC dev from my laptop, the order was created in CMR db. (although it was not processed due to firewall issues for obvious reasons)  
* Delayed_job workers are busy in OPS  
* Delayed_job “queue” has a length of 365 (Delayed::Job.where(attempts: 0).size)  
* Every time the user submits an order, the order GUID is different, meaning that the queue of the delayed_job is moving.  
* When the queue is empty, the worker will log something like: 1 jobs processed at 8.5161 j/s, 0 failed. Splunk search sourcetype=prod*ngap*app_search-ops "jobs processed at” shows that the queue has never been empty starting from Monday. Also note that the processing time is 8.5s per job (I admit there are faster ones but the time varies greatly from provider to provider). If that is the case for LAADS or if the processing time is larger than the interval that the user submits, the queue will get longer and longer. 365 is the number from today around 3pm.
* All the findings are pointing to the delayed job queue being to long. However, I am not 100% sure that cleaning up the queue will resolve this issue.

I suggest to recycle both ops nodes first and take another look at the delayed_jobs status:  

A) Find all rows in edsc db: Delayed::Job.all.size  

B) Find queued rows (i.e. the size of the queue): Delayed::Job.where(attempts: 0).size  

C) Find failed jobs (which are permanently kept in the db; successful ones are immediately removed from the db upon completion): Delayed::Job.where.not(failed_at: nil).size  

A) = B) + C)

Only clean up the queue if the size gets larger over time.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

The short term solution is 

* purge the db
* notify the user to stop submitting repeated orders to EDSC
* add status banner to indicate that LAADS ordering is down

Long term: we could increase the number of workers to work off the queue faster. (in file: earthdata-search/Procfile, add -n=2 before run in the second line)

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

From [~tgoff]:

It is speculation, however, we know this:

1) LAADS AND USGS_EROS provider orders are being suspended due to the firwall issue.

2) Orders for other providers are getting through

3) Orders submitted via OPS EDSC do not make it to legacy services(in a reasonable time period), but orders from OPS reverb or UAT EDSC pointed at OPS do

4) Orders submitted via OPS EDSC do eventually make it to legacy services, after a number of hours  

5) A single user is submitting an unusual volume of orders for LAADS data

Based on this, I think the best explanation we have is that the queue is being exhausted by the LAADS orders from a single user, and the suspension of LAADS order is contributing to its slower workoff (though Im not sure how this would happen, unless EDSC is checking on the status of orders it submits and only allowing a certain number to be in progress at a time).

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[~dnewman], do you concur that this ticket is no longer needed?

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673

**Comment by earthdata-tasktop** [comment]
(Comment from Frank Schaffer):

This issue is affecting both NSIDC orders and EGI requests in Prod.

**Comment by earthdata-tasktop** [comment]
(Comment from Sudha Murthy):

saw the issue in all three daacs - NSIDC/LPDAAC/LARC where the order was stuck in 'Creating'

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

What caused the delayed queue backup?

* We don’t know what caused the backup. We suspect that it is one user who kept submitting orders around the clock since this Monday using collection C203234448-LAADS. Starting from 5/8 4:12pm to 7pm, the user submitted one order approximately every 5 minutes. From 7pm, the user brought up another instance to submit one order every 5 minutes and have been doing that since then. 
* Doug saw the issue this morning and Tim and I started investigating. Here are what we found:  
* LAADS has a firewall issue and all their orders are blocked. From the EDSC’s point of view, the order stuck in “Creating” state. (The one that the user used is a LAADS collection)  
* Using an NSIDC collection and submitting a new order in EDSC OPS, the order was not created in CMR db.  
* Using an NSIDC collection and submitting a new order in EDSC UAT pointing to CMR OPS (cmr_env=prod), the order reached CMR and was processed correctly.  
* Using an NSIDC collection and submitting a new order in EDSC dev from my laptop, the order was created in CMR db. (although it was not processed due to firewall issues for obvious reasons)  
* Delayed_job workers are busy in OPS  
* Delayed_job “queue” has a length of 365 (Delayed::Job.where(attempts: 0).size)  
* Every time the user submits an order, the order GUID is different, meaning that the queue of the delayed_job is moving.  
* When the queue is empty, the worker will log something like: 1 jobs processed at 8.5161 j/s, 0 failed. Splunk search sourcetype=prod*ngap*app_search-ops "jobs processed at” shows that the queue has never been empty starting from Monday. Also note that the processing time is 8.5s per job (I admit there are faster ones but the time varies greatly from provider to provider). If that is the case for LAADS or if the processing time is larger than the interval that the user submits, the queue will get longer and longer. 365 is the number from today around 3pm.
* All the findings are pointing to the delayed job queue being to long. However, I am not 100% sure that cleaning up the queue will resolve this issue.

I suggest to recycle both ops nodes first and take another look at the delayed_jobs status:  

A) Find all rows in edsc db: Delayed::Job.all.size  

B) Find queued rows (i.e. the size of the queue): Delayed::Job.where(attempts: 0).size  

C) Find failed jobs (which are permanently kept in the db; successful ones are immediately removed from the db upon completion): Delayed::Job.where.not(failed_at: nil).size  

A) = B) + C)

Only clean up the queue if the size gets larger over time.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

The short term solution is 

* purge the db
* notify the user to stop submitting repeated orders to EDSC
* add status banner to indicate that LAADS ordering is down

Long term: we could increase the number of workers to work off the queue faster. (in file: earthdata-search/Procfile, add -n=2 before run in the second line)

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

From [~tgoff]:

It is speculation, however, we know this:

1) LAADS AND USGS_EROS provider orders are being suspended due to the firwall issue.

2) Orders for other providers are getting through

3) Orders submitted via OPS EDSC do not make it to legacy services(in a reasonable time period), but orders from OPS reverb or UAT EDSC pointed at OPS do

4) Orders submitted via OPS EDSC do eventually make it to legacy services, after a number of hours  

5) A single user is submitting an unusual volume of orders for LAADS data

Based on this, I think the best explanation we have is that the queue is being exhausted by the LAADS orders from a single user, and the suspension of LAADS order is contributing to its slower workoff (though Im not sure how this would happen, unless EDSC is checking on the status of orders it submits and only allowing a certain number to be in progress at a time).

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[~dnewman], do you concur that this ticket is no longer needed?

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673

**Comment by earthdata-tasktop** [comment]
(Comment from Sudha Murthy):

saw the issue in all three daacs - NSIDC/LPDAAC/LARC where the order was stuck in 'Creating'

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

What caused the delayed queue backup?

* We don’t know what caused the backup. We suspect that it is one user who kept submitting orders around the clock since this Monday using collection C203234448-LAADS. Starting from 5/8 4:12pm to 7pm, the user submitted one order approximately every 5 minutes. From 7pm, the user brought up another instance to submit one order every 5 minutes and have been doing that since then. 
* Doug saw the issue this morning and Tim and I started investigating. Here are what we found:  
* LAADS has a firewall issue and all their orders are blocked. From the EDSC’s point of view, the order stuck in “Creating” state. (The one that the user used is a LAADS collection)  
* Using an NSIDC collection and submitting a new order in EDSC OPS, the order was not created in CMR db.  
* Using an NSIDC collection and submitting a new order in EDSC UAT pointing to CMR OPS (cmr_env=prod), the order reached CMR and was processed correctly.  
* Using an NSIDC collection and submitting a new order in EDSC dev from my laptop, the order was created in CMR db. (although it was not processed due to firewall issues for obvious reasons)  
* Delayed_job workers are busy in OPS  
* Delayed_job “queue” has a length of 365 (Delayed::Job.where(attempts: 0).size)  
* Every time the user submits an order, the order GUID is different, meaning that the queue of the delayed_job is moving.  
* When the queue is empty, the worker will log something like: 1 jobs processed at 8.5161 j/s, 0 failed. Splunk search sourcetype=prod*ngap*app_search-ops "jobs processed at” shows that the queue has never been empty starting from Monday. Also note that the processing time is 8.5s per job (I admit there are faster ones but the time varies greatly from provider to provider). If that is the case for LAADS or if the processing time is larger than the interval that the user submits, the queue will get longer and longer. 365 is the number from today around 3pm.
* All the findings are pointing to the delayed job queue being to long. However, I am not 100% sure that cleaning up the queue will resolve this issue.

I suggest to recycle both ops nodes first and take another look at the delayed_jobs status:  

A) Find all rows in edsc db: Delayed::Job.all.size  

B) Find queued rows (i.e. the size of the queue): Delayed::Job.where(attempts: 0).size  

C) Find failed jobs (which are permanently kept in the db; successful ones are immediately removed from the db upon completion): Delayed::Job.where.not(failed_at: nil).size  

A) = B) + C)

Only clean up the queue if the size gets larger over time.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

The short term solution is 

* purge the db
* notify the user to stop submitting repeated orders to EDSC
* add status banner to indicate that LAADS ordering is down

Long term: we could increase the number of workers to work off the queue faster. (in file: earthdata-search/Procfile, add -n=2 before run in the second line)

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

From [~tgoff]:

It is speculation, however, we know this:

1) LAADS AND USGS_EROS provider orders are being suspended due to the firwall issue.

2) Orders for other providers are getting through

3) Orders submitted via OPS EDSC do not make it to legacy services(in a reasonable time period), but orders from OPS reverb or UAT EDSC pointed at OPS do

4) Orders submitted via OPS EDSC do eventually make it to legacy services, after a number of hours  

5) A single user is submitting an unusual volume of orders for LAADS data

Based on this, I think the best explanation we have is that the queue is being exhausted by the LAADS orders from a single user, and the suspension of LAADS order is contributing to its slower workoff (though Im not sure how this would happen, unless EDSC is checking on the status of orders it submits and only allowing a certain number to be in progress at a time).

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[~dnewman], do you concur that this ticket is no longer needed?

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

What caused the delayed queue backup?

* We don’t know what caused the backup. We suspect that it is one user who kept submitting orders around the clock since this Monday using collection C203234448-LAADS. Starting from 5/8 4:12pm to 7pm, the user submitted one order approximately every 5 minutes. From 7pm, the user brought up another instance to submit one order every 5 minutes and have been doing that since then. 
* Doug saw the issue this morning and Tim and I started investigating. Here are what we found:  
* LAADS has a firewall issue and all their orders are blocked. From the EDSC’s point of view, the order stuck in “Creating” state. (The one that the user used is a LAADS collection)  
* Using an NSIDC collection and submitting a new order in EDSC OPS, the order was not created in CMR db.  
* Using an NSIDC collection and submitting a new order in EDSC UAT pointing to CMR OPS (cmr_env=prod), the order reached CMR and was processed correctly.  
* Using an NSIDC collection and submitting a new order in EDSC dev from my laptop, the order was created in CMR db. (although it was not processed due to firewall issues for obvious reasons)  
* Delayed_job workers are busy in OPS  
* Delayed_job “queue” has a length of 365 (Delayed::Job.where(attempts: 0).size)  
* Every time the user submits an order, the order GUID is different, meaning that the queue of the delayed_job is moving.  
* When the queue is empty, the worker will log something like: 1 jobs processed at 8.5161 j/s, 0 failed. Splunk search sourcetype=prod*ngap*app_search-ops "jobs processed at” shows that the queue has never been empty starting from Monday. Also note that the processing time is 8.5s per job (I admit there are faster ones but the time varies greatly from provider to provider). If that is the case for LAADS or if the processing time is larger than the interval that the user submits, the queue will get longer and longer. 365 is the number from today around 3pm.
* All the findings are pointing to the delayed job queue being to long. However, I am not 100% sure that cleaning up the queue will resolve this issue.

I suggest to recycle both ops nodes first and take another look at the delayed_jobs status:  

A) Find all rows in edsc db: Delayed::Job.all.size  

B) Find queued rows (i.e. the size of the queue): Delayed::Job.where(attempts: 0).size  

C) Find failed jobs (which are permanently kept in the db; successful ones are immediately removed from the db upon completion): Delayed::Job.where.not(failed_at: nil).size  

A) = B) + C)

Only clean up the queue if the size gets larger over time.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

The short term solution is 

* purge the db
* notify the user to stop submitting repeated orders to EDSC
* add status banner to indicate that LAADS ordering is down

Long term: we could increase the number of workers to work off the queue faster. (in file: earthdata-search/Procfile, add -n=2 before run in the second line)

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

From [~tgoff]:

It is speculation, however, we know this:

1) LAADS AND USGS_EROS provider orders are being suspended due to the firwall issue.

2) Orders for other providers are getting through

3) Orders submitted via OPS EDSC do not make it to legacy services(in a reasonable time period), but orders from OPS reverb or UAT EDSC pointed at OPS do

4) Orders submitted via OPS EDSC do eventually make it to legacy services, after a number of hours  

5) A single user is submitting an unusual volume of orders for LAADS data

Based on this, I think the best explanation we have is that the queue is being exhausted by the LAADS orders from a single user, and the suspension of LAADS order is contributing to its slower workoff (though Im not sure how this would happen, unless EDSC is checking on the status of orders it submits and only allowing a certain number to be in progress at a time).

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[~dnewman], do you concur that this ticket is no longer needed?

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

The short term solution is 

* purge the db
* notify the user to stop submitting repeated orders to EDSC
* add status banner to indicate that LAADS ordering is down

Long term: we could increase the number of workers to work off the queue faster. (in file: earthdata-search/Procfile, add -n=2 before run in the second line)

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

From [~tgoff]:

It is speculation, however, we know this:

1) LAADS AND USGS_EROS provider orders are being suspended due to the firwall issue.

2) Orders for other providers are getting through

3) Orders submitted via OPS EDSC do not make it to legacy services(in a reasonable time period), but orders from OPS reverb or UAT EDSC pointed at OPS do

4) Orders submitted via OPS EDSC do eventually make it to legacy services, after a number of hours  

5) A single user is submitting an unusual volume of orders for LAADS data

Based on this, I think the best explanation we have is that the queue is being exhausted by the LAADS orders from a single user, and the suspension of LAADS order is contributing to its slower workoff (though Im not sure how this would happen, unless EDSC is checking on the status of orders it submits and only allowing a certain number to be in progress at a time).

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[~dnewman], do you concur that this ticket is no longer needed?

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

From [~tgoff]:

It is speculation, however, we know this:

1) LAADS AND USGS_EROS provider orders are being suspended due to the firwall issue.

2) Orders for other providers are getting through

3) Orders submitted via OPS EDSC do not make it to legacy services(in a reasonable time period), but orders from OPS reverb or UAT EDSC pointed at OPS do

4) Orders submitted via OPS EDSC do eventually make it to legacy services, after a number of hours  

5) A single user is submitting an unusual volume of orders for LAADS data

Based on this, I think the best explanation we have is that the queue is being exhausted by the LAADS orders from a single user, and the suspension of LAADS order is contributing to its slower workoff (though Im not sure how this would happen, unless EDSC is checking on the status of orders it submits and only allowing a certain number to be in progress at a time).

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[~dnewman], do you concur that this ticket is no longer needed?

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[~dnewman], do you concur that this ticket is no longer needed?

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

What caused the delayed queue backup?

* We don’t know what caused the backup. We suspect that it is one user who kept submitting orders around the clock since this Monday using collection C203234448-LAADS. Starting from 5/8 4:12pm to 7pm, the user submitted one order approximately every 5 minutes. From 7pm, the user brought up another instance to submit one order every 5 minutes and have been doing that since then. 
* Doug saw the issue this morning and Tim and I started investigating. Here are what we found:  
* LAADS has a firewall issue and all their orders are blocked. From the EDSC’s point of view, the order stuck in “Creating” state. (The one that the user used is a LAADS collection)  
* Using an NSIDC collection and submitting a new order in EDSC OPS, the order was not created in CMR db.  
* Using an NSIDC collection and submitting a new order in EDSC UAT pointing to CMR OPS (cmr_env=prod), the order reached CMR and was processed correctly.  
* Using an NSIDC collection and submitting a new order in EDSC dev from my laptop, the order was created in CMR db. (although it was not processed due to firewall issues for obvious reasons)  
* Delayed_job workers are busy in OPS  
* Delayed_job “queue” has a length of 365 (Delayed::Job.where(attempts: 0).size)  
* Every time the user submits an order, the order GUID is different, meaning that the queue of the delayed_job is moving.  
* When the queue is empty, the worker will log something like: 1 jobs processed at 8.5161 j/s, 0 failed. Splunk search sourcetype=prod*ngap*app_search-ops "jobs processed at” shows that the queue has never been empty starting from Monday. Also note that the processing time is 8.5s per job (I admit there are faster ones but the time varies greatly from provider to provider). If that is the case for LAADS or if the processing time is larger than the interval that the user submits, the queue will get longer and longer. 365 is the number from today around 3pm.
* All the findings are pointing to the delayed job queue being to long. However, I am not 100% sure that cleaning up the queue will resolve this issue.

I suggest to recycle both ops nodes first and take another look at the delayed_jobs status:  

A) Find all rows in edsc db: Delayed::Job.all.size  

B) Find queued rows (i.e. the size of the queue): Delayed::Job.where(attempts: 0).size  

C) Find failed jobs (which are permanently kept in the db; successful ones are immediately removed from the db upon completion): Delayed::Job.where.not(failed_at: nil).size  

A) = B) + C)

Only clean up the queue if the size gets larger over time.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

The short term solution is 

* purge the db
* notify the user to stop submitting repeated orders to EDSC
* add status banner to indicate that LAADS ordering is down

Long term: we could increase the number of workers to work off the queue faster. (in file: earthdata-search/Procfile, add -n=2 before run in the second line)

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

From [~tgoff]:

It is speculation, however, we know this:

1) LAADS AND USGS_EROS provider orders are being suspended due to the firwall issue.

2) Orders for other providers are getting through

3) Orders submitted via OPS EDSC do not make it to legacy services(in a reasonable time period), but orders from OPS reverb or UAT EDSC pointed at OPS do

4) Orders submitted via OPS EDSC do eventually make it to legacy services, after a number of hours  

5) A single user is submitting an unusual volume of orders for LAADS data

Based on this, I think the best explanation we have is that the queue is being exhausted by the LAADS orders from a single user, and the suspension of LAADS order is contributing to its slower workoff (though Im not sure how this would happen, unless EDSC is checking on the status of orders it submits and only allowing a certain number to be in progress at a time).

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[~dnewman], do you concur that this ticket is no longer needed?

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

The short term solution is 

* purge the db
* notify the user to stop submitting repeated orders to EDSC
* add status banner to indicate that LAADS ordering is down

Long term: we could increase the number of workers to work off the queue faster. (in file: earthdata-search/Procfile, add -n=2 before run in the second line)

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

From [~tgoff]:

It is speculation, however, we know this:

1) LAADS AND USGS_EROS provider orders are being suspended due to the firwall issue.

2) Orders for other providers are getting through

3) Orders submitted via OPS EDSC do not make it to legacy services(in a reasonable time period), but orders from OPS reverb or UAT EDSC pointed at OPS do

4) Orders submitted via OPS EDSC do eventually make it to legacy services, after a number of hours  

5) A single user is submitting an unusual volume of orders for LAADS data

Based on this, I think the best explanation we have is that the queue is being exhausted by the LAADS orders from a single user, and the suspension of LAADS order is contributing to its slower workoff (though Im not sure how this would happen, unless EDSC is checking on the status of orders it submits and only allowing a certain number to be in progress at a time).

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[~dnewman], do you concur that this ticket is no longer needed?

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

From [~tgoff]:

It is speculation, however, we know this:

1) LAADS AND USGS_EROS provider orders are being suspended due to the firwall issue.

2) Orders for other providers are getting through

3) Orders submitted via OPS EDSC do not make it to legacy services(in a reasonable time period), but orders from OPS reverb or UAT EDSC pointed at OPS do

4) Orders submitted via OPS EDSC do eventually make it to legacy services, after a number of hours  

5) A single user is submitting an unusual volume of orders for LAADS data

Based on this, I think the best explanation we have is that the queue is being exhausted by the LAADS orders from a single user, and the suspension of LAADS order is contributing to its slower workoff (though Im not sure how this would happen, unless EDSC is checking on the status of orders it submits and only allowing a certain number to be in progress at a time).

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[~dnewman], do you concur that this ticket is no longer needed?

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

[~dnewman], do you concur that this ticket is no longer needed?

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673

**Comment by earthdata-tasktop** [comment]
(Comment from Mark Reese):

Blocked by NGAP-1673
